### PR TITLE
feat(commands): add descriptions

### LINKED ISSUE #2076: Add descriptions to commands via `vim.api.nvim_create_user_command`
**Is your feature request related to a problem? Please describe.**

Trying to build a good `legendary.nvim` extension for `nvim-tree.lua`

**Describe the solution you'd like**

You create the commands here: https://github.com/nvim-tree/nvim-tree.lua/blob/aa9971768a08caa4f10f94ab84e48d2ceb30b1c0/lua/nvim-tree.lua#L245-L280

The `vim.api.nvim_create_user_command` function accepts a `desc` field to provide a description that will be shown if you look up the commands via the Neovim API.

**Describe alternatives you've considered**

Hard-coding the descriptions.

**Additional context**

https://github.com/mrjones2014/legendary.nvim/pull/320

### PR DESCRIPTION
**Pull request by alex-courtis**
fixes #2076 

I would be most grateful for your review @mrjones2014

I've used the `"nvim-tree: "` prefix as per the mappings and kept it brief. Most grateful for suggestions...

### CODE REVIEW

File: lua/nvim-tree.lua [hunk 1]
```diff
@@ -245,38 +245,53 @@ end
...
   vim.api.nvim_create_user_command("NvimTreeCollapseKeepBuffers", function()
     require("nvim-tree.api").tree.collapse_all(true)
-  end, { bar = true })
-  vim.api.nvim_create_user_command("NvimTreeGenerateOnAttach", keymap_legacy.cmd_generate_on_attach, {})
+  end, { bar = true, desc = "nvim-tree: collapse, keep directories open" })
+
+  vim.api.nvim_create_user_command(
+    "NvimTreeGenerateOnAttach",
+    keymap_legacy.cmd_generate_on_attach,
+    { desc = "nvim-tree: generate on_attach from view.mappings" }
```
Comment (gegoune) [inline lua/nvim-tree.lua:293]:
```suggestion
    { desc = "nvim-tree: generate on_attach function from deprecated view.mappings" }
```

**Review by gegoune — COMMENTED** [review]
Many of those descriptions just repeats what the command itself already says. This doesn't add any value really. But frankly I have no idea what else it could say. :)

**Comment by mrjones2014** [comment]
Descriptions look good. Is there any way you can expose the commands as raw data that we can access directly? Sorta like this: https://github.com/mrjones2014/op.nvim/blob/master/lua/op/commands.lua

File: lua/nvim-tree.lua (hunk 1 above)
Comment (alex-courtis) [reply lua/nvim-tree.lua]: I like the suggestion feature. That will make life a lot easier.

**Comment by alex-courtis** [comment]
> Descriptions look good. Is there any way you can expose the commands as raw data that we can access directly? Sorta like this: https://github.com/mrjones2014/op.nvim/blob/master/lua/op/commands.lua

That's a great idea. Might be a good opportunity to extract the commands to their own file.

**Comment by alex-courtis** [comment]
@mrjones2014  I would be most grateful if you tested the new `api.get_commands` as well as testing each of the commands, to ensure that I didn't miss anything in the refactor.

**Comment by gegoune** [comment]
Would we ever have any more endpoints for commands? If there is a slightest change maybe it warrants nesting like `api.commands.get()`?

**Review by mrjones2014 — APPROVED** [review]
As long as the table structure remains consistent, looks good to me 👍

**Comment by alex-courtis** [comment]
> Would we ever have any more endpoints for commands? If there is a slightest change maybe it warrants nesting like `api.commands.get()`?

Much better. Everything is nested now.

**Pull request merged**